### PR TITLE
feat: Add C make/cmake application pipelines

### PR DESCRIPTION
**Pull request by Rolika4**
## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce.

## Checklist:
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Pull Request contain one commit. I squash my commits.

### CODE REVIEW

File: charts/pipelines-library/templates/tasks/edp-c.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+spec:
+  description: >-
+    This task can be used to run npm goals on a project
+    where package.json is present and has some pre-defined
+    npm scripts.
+  params:
+    - name: PATH_CONTEXT
+      type: string
+      default: "source"
+      description: The path where package.json of the project is defined.
```
Comment (SergK) [inline charts/pipelines-library/templates/tasks/edp-c.yaml:21]: are we ok with this description?

File: charts/pipelines-library/templates/tasks/edp-c.yaml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    - name: BASE_IMAGE
+      type: string
+      default: "golang:1.22-bookworm"
```
Comment (SergK) [inline charts/pipelines-library/templates/tasks/edp-c.yaml:24]: are we ok with this image?

**Review by SergK — CHANGES_REQUESTED** [review]

File: charts/pipelines-library/templates/tasks/edp-c.yaml (hunk 1 above)
Comment (Rolika4) [reply charts/pipelines-library/templates/tasks/edp-c.yaml]: Update description

File: charts/pipelines-library/templates/tasks/edp-c.yaml (hunk 2 above)
Comment (Rolika4) [reply charts/pipelines-library/templates/tasks/edp-c.yaml]: Update image

**Review by SergK — APPROVED** [review]

**Pull request merged**